### PR TITLE
fix command split on Windows

### LINKED ISSUE #198: FileNotFoundError when using run_process - Windows
### Description

I am trying to watch a directory but `watchfiles` throws me `FileNotFoundError`

After some digging I found the root cause
the [`shlex.split`](https://docs.python.org/3/library/shlex.html#shlex.split) produces `E:\\myapp\\main.py` as `['python', 'E:myappmain.py']` and It works fine when using `shlex.split(cmd, posix=False)`

It only occurs on windows.
I tried to reproduce this on my linux machine, there it works fine

### Example Code

```Python
import watchfiles

print("Version:", watchfiles.__version__)

watchfiles.run_process("E:\\myapp", target="python E:\\myapp\\main.py", target_type="command", debug=True)
```

### Watchfiles Output

```Text
Version: 0.18.0
watcher: ReadDirectoryChangesWatcher { tx: Sender { .. }, cmd_rx: Receiver { .. }, wakeup_sem: 0x24c }
python: can't open file 'E:myappmain.py': [Errno 2] No such file or directory
```

### Operating System & Architecture

Windows-10-10.0.19041-SP0
10.0.19041

### Python & Watchfiles Version

python 3.7.7
watchfiles: 0.18.0

### PR DESCRIPTION
**Pull request by Kulothungan16**

### CODE REVIEW

File: watchfiles/run.py [hunk 1]
```diff
@@ -250,7 +250,7 @@ def start_process(
             logger.warning('ignoring args and kwargs for "command" target')
 
         assert isinstance(target, str), 'target must be a string to run as a command'
-        popen_args = shlex.split(target)
+        popen_args = shlex.split(target, posix=False)
```
Comment (samuelcolvin) [inline watchfiles/run.py:253]:
```suggestion
        import platform

        posix = platform.uname().system.lower() != 'windows'
        popen_args = shlex.split(target, posix=posix)
```
surely?

Comment (samuelcolvin) [reply watchfiles/run.py:253]: might be best to move this into a separate function for ease of testing.

**Review by samuelcolvin — COMMENTED** [review]
Thanks for this, we also need tests for this.

File: watchfiles/run.py (hunk 1 above)
Comment (Kulothungan16) [reply watchfiles/run.py]: Made a new function for that...
Could you write test code for the split_cmd function, I don't quite understand the testings

Thanks 😃

File: watchfiles/run.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 import json
 import logging
 import os
+import platform
```
Comment (samuelcolvin) [inline watchfiles/run.py:5]: local import please, platform might be quite slow to import.

File: watchfiles/run.py [hunk 3]
```diff
@@ -219,6 +220,11 @@ async def main():
 spawn_context = get_context('spawn')
 
 
+def split_cmd(cmd: str) -> list[str]:
```
Comment (samuelcolvin) [inline watchfiles/run.py:223]:
```suggestion
def split_cmd(cmd: str) -> List[str]:
```

**Review by samuelcolvin — COMMENTED** [review]
Still need tests.

File: watchfiles/run.py (hunk 1 above)
Comment (samuelcolvin) [reply watchfiles/run.py]: No, you've created the PR, you should write the tests.

I don't use windows, I don't need this.

**Comment by samuelcolvin** [comment]
This project uses single quotes, it's worth setting up linting, pre-commit before committing to avoid these issues.

File: tests/test_run_process.py [hunk 4]
```diff
@@ -68,6 +68,19 @@ def test_dead_callback(mocker, mock_rust_notify: 'MockRustType'):
     c.assert_called_with({(Change.added, '/path/to/foobar.py')})
 
 
+@pytest.mark.skipif(sys.platform in ['linux', 'darwin'], reason='no need to test this on linux or mac osx')
```
Comment (samuelcolvin) [inline tests/test_run_process.py:71]:
```suggestion
@pytest.mark.skipif(sys.platform != 'win32', reason='no need to test this except on windows')
```

**Review by samuelcolvin — COMMENTED** [review]
otherwise, this looks great. Thank you for the contribution.

**Comment by samuelcolvin** [comment]
Thanks so much.

**Comment by Kulothungan16** [comment]
Thanks for the assistance through the entire process 😃

A release containing this fix would be helpful 🙏

**Comment by samuelcolvin** [comment]
> A release containing this fix would be helpful pray

Will do next week, if I forget, please prompt me again.

**Comment by Kulothungan16** [comment]
Fine, thanks

**Comment by Kulothungan16** [comment]
> Will do next week, if I forget, please prompt me again.

ping

**Pull request merged**